Use Expected for errors in WebExtension

ItsJamie9494 opened the pull request:
#### 7a2430f73eb2e5fe6adc7ebc5fc789a6ab4cd0c9
<pre>
Use Expected for errors in WebExtension
<a href="https://bugs.webkit.org/show_bug.cgi?id=297831">https://bugs.webkit.org/show_bug.cgi?id=297831</a>

Reviewed by Timothy Hatcher.

Follow-up to 299009@main, use Expected&lt;bool, RefPtr&lt;API::Error&gt;&gt; or the appropriate return type in WebExtension,
dropping outError. As well, make sure to passthrough the underlyingError when creating new API::Errors.

* Source/WebKit/Shared/Extensions/WebExtensionLocalization.cpp:
(WebKit::WebExtensionLocalization::localizationJSONForWebExtension):
* Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIScriptingCocoa.mm:
(WebKit::WebExtensionContext::createInjectedContentForScripts):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionCocoa.mm:
(WebKit::WebExtension::resourceDataForPath):
(WebKit::WebExtension::iconForPath):
(WebKit::WebExtension::bestIcon):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm:
(WebKit::WebExtensionContext::addInjectedContent):
(WebKit::WebExtensionContext::loadDeclarativeNetRequestRules):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionDynamicScriptsCocoa.mm:
(WebKit::WebExtensionDynamicScripts::sourcePairForResource):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionURLSchemeHandlerCocoa.mm:
(WebKit::WebExtensionURLSchemeHandler::platformStartTask):
* Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
(WebKit::WebExtension::manifestObject):
(WebKit::WebExtension::resourceStringForPath):
(WebKit::WebExtension::createError):
(WebKit::WebExtension::populateActionPropertiesIfNeeded):
(WebKit::WebExtension::parseDeclarativeNetRequestRulesetObject):
(WebKit::WebExtension::populateDeclarativeNetRequestPropertiesIfNeeded):
* Source/WebKit/UIProcess/Extensions/WebExtension.h:
(WebKit::WebExtension::iconForPath):

Canonical link: <a href="https://commits.webkit.org/300035@main">https://commits.webkit.org/300035@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d167d88f86893fe312b20efb3d476901bab2a93a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/121078 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/40774 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/31432 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/127497 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/73159 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/41476 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/49353 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/91969 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/61184 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/124030 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/33117 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/108535 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/72653 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/32141 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/71088 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/102623 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/26817 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/130350 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/48005 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/36483 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/100578 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/48373 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/104702 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/100480 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/45883 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/23935 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/44697 "The change is no longer eligible for processing.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19210 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/47863 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/53576 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/47334 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/50681 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/49018 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

xeenon commented on Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionDynamicScriptsCocoa.mm:
```diff
@@ -89,14 +89,15 @@ static bool userStyleSheetMatchesContent(Ref<API::UserStyleSheet> userStyleSheet
         return std::nullopt;
     }
 
+    auto scriptString = scriptStringResult.value();
+
```
```suggestion
    auto scriptString = scriptStringResult.value();
```

ItsJamie9494 commented on Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionCocoa.mm:
```diff
@@ -226,65 +224,64 @@
-            outError = createError(Error::ResourceNotFound, WEB_UI_FORMAT_CFSTRING("Unable to find \"%@\" in the extension’s resources. It is an invalid path.", "WKWebExtensionErrorResourceNotFound description with invalid file path", bridge_cast(cocoaPath)));
-        return nullptr;
+            return makeUnexpected(createError(Error::ResourceNotFound, WEB_UI_FORMAT_CFSTRING("Unable to find \"%@\" in the extension’s resources. It is an invalid path.", "WKWebExtensionErrorResourceNotFound description with invalid file path", bridge_cast(cocoaPath))));
+        return makeUnexpected(RefPtr<API::Error>(nullptr));
```
I tried that initially, but if I recall correctly there were issues about the type not matching properly

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/7d1dccfb01bec868470f31fb3f9449317244f52b. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49820

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/48e886bf7fe4f9d003d69d8b690c0d4d5e9c9150. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49820

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d167d88f86893fe312b20efb3d476901bab2a93a. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49820

ItsJamie9494 committed:
Use Expected for errors in WebExtension
https://bugs.webkit.org/show_bug.cgi?id=297831

Reviewed by Timothy Hatcher.

Follow-up to 299009@main, use Expected<bool, RefPtr<API::Error>> or the appropriate return type in WebExtension,
dropping outError. As well, make sure to passthrough the underlyingError when creating new API::Errors.

* Source/WebKit/Shared/Extensions/WebExtensionLocalization.cpp:
(WebKit::WebExtensionLocalization::localizationJSONForWebExtension):
* Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIScriptingCocoa.mm:
(WebKit::WebExtensionContext::createInjectedContentForScripts):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionCocoa.mm:
(WebKit::WebExtension::resourceDataForPath):
(WebKit::WebExtension::iconForPath):
(WebKit::WebExtension::bestIcon):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm:
(WebKit::WebExtensionContext::addInjectedContent):
(WebKit::WebExtensionContext::loadDeclarativeNetRequestRules):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionDynamicScriptsCocoa.mm:
(WebKit::WebExtensionDynamicScripts::sourcePairForResource):
* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionURLSchemeHandlerCocoa.mm:
(WebKit::WebExtensionURLSchemeHandler::platformStartTask):
* Source/WebKit/UIProcess/Extensions/WebExtension.cpp:
(WebKit::WebExtension::manifestObject):
(WebKit::WebExtension::resourceStringForPath):
(WebKit::WebExtension::createError):
(WebKit::WebExtension::populateActionPropertiesIfNeeded):
(WebKit::WebExtension::parseDeclarativeNetRequestRulesetObject):
(WebKit::WebExtension::populateDeclarativeNetRequestPropertiesIfNeeded):
* Source/WebKit/UIProcess/Extensions/WebExtension.h:
(WebKit::WebExtension::iconForPath):

Canonical link: https://commits.webkit.org/300035@main

webkit-commit-queue commented:
Committed 300035@main (7a2430f73eb2): <https://commits.webkit.org/300035@main>

Reviewed commits have been landed. Closing PR #49820 and removing active labels.
